client.go

```diff
@@ -37,21 +38,21 @@ func WithLogger(l logging.Logger) ClientOption {
 }
 
 // WithImageFactory supply your own image factory.
-func WithImageFactory(f ImageFactory) ClientOption {
+func WithImageFactory(f internal.ImageFactory) ClientOption {
 	return func(c *Client) {
 		c.imageFactory = f
 	}
 }
 
 // WithFetcher supply your own fetcher.
-func WithFetcher(f ImageFetcher) ClientOption {
+func WithFetcher(f internal.ImageFetcher) ClientOption {
 	return func(c *Client) {
 		c.imageFetcher = f
 	}
 }
 
 // WithDownloader supply your own downloader.
-func WithDownloader(d Downloader) ClientOption {
+func WithDownloader(d internal.Downloader) ClientOption {
```

Comment:
A general rule should be that anything _used_ by a public API cannot reference an internal component. This is an example of where that is violated. Another example would be returning an internal object/interface from a public method.

Reply:
I agree, and missed this. I also missed that the package here ended up as `internal`, which I believe is not recommended.

Reply:
Fixed.